cmd/shipper-state-metrics/collector.go

```diff
@@ -322,6 +332,36 @@ func (ssm ShipperStateMetrics) collectClusters(ch chan<- prometheus.Metric) {
 	}
 }
 
+func (ssm ShipperStateMetrics) collectRolloutBlocks(ch chan<- prometheus.Metric) {
+	nss, err := getNamespaces(ssm.nssLister)
+	if err != nil {
+		glog.Errorf("collect Namespaces: %s", err)
+		return
+	}
+
+	rolloutBlocks, err := ssm.rbLister.List(everything)
+	if err != nil {
+		glog.Errorf("collect RolloutBlocks: %s", err)
+		return
+	}
+
+	rbsPerNamespace := make(map[string]float64)
+	for _, rolloutBlock := range rolloutBlocks {
+		rbsPerNamespace[rolloutBlock.Namespace]++
+	}
+
+	glog.V(4).Infof("RolloutBlocks: %v", rbsPerNamespace)
+
+	for _, ns := range nss {
+		n, ok := rbsPerNamespace[ns.Name]
+		if !ok {
+			n = 0
+		}
```

Comment:
Minor: not needed: you can rely on golang map zero-value return, in this case it's going to be 0.0